static/app/views/issueList/draggableTabMenuButton.tsx

```diff
@@ -0,0 +1,164 @@
+   * Note: The `Discard Changes` button only appears for persistent tabs when `isChanged=true`
+   */
+  onDiscard?: (key: MenuItemProps['key']) => void;
+
+  /**
+   * Callback function to be called when user clicks the 'Duplicate' Button
+   * Note: The `Duplicate` button only appears when `isTempTab=false` (persistent tabs)
+   */
+  onDuplicate?: (key: MenuItemProps['key']) => void;
+
+  /**
+   * Callback function to be called when user clicks the 'Rename' Button
+   * Note: The `Rename` button only appears when `isTempTab=false` (persistent tabs)
+   * @returns
+   */
+  onRename?: (key: MenuItemProps['key']) => void;
+  /**
+   * Callback function to be called when user clicks the 'Save' button.
+   * Note: The `Save` button only appears for persistent tabs when `isChanged=true`, or when `isTempTab=true`
+   */
+  onSave?: (key: MenuItemProps['key']) => void;
+  triggerProps?: Omit<React.HTMLAttributes<HTMLElement>, 'children'>;
+}
+
+export function DraggableTabMenuButton({
+  triggerProps,
+  hasUnsavedChanges = false,
+  onDelete,
+  onDiscard,
+  onDuplicate,
+  onRename,
+  onSave,
+}: DraggableTabMenuButtonProps) {
+  const hasUnsavedChangesMenuOptions: MenuItemProps[] = [
+    {
+      key: 'save-changes',
+      label: t('Save Changes'),
+      priority: 'primary',
+      onAction: onSave,
+    },
+    {
+      key: 'discard-changes',
+      label: t('Discard Changes'),
+      onAction: onDiscard,
+    },
+  ];
+
+  const defaultMenuOptions: MenuItemProps[] = [
+    {
+      key: 'rename-tab',
+      label: t('Rename'),
+      onAction: onRename,
+    },
+    {
+      key: 'duplicate-tab',
+      label: t('Duplicate'),
+      onAction: onDuplicate,
+    },
+    {
+      key: 'delete-tab',
+      label: t('Delete'),
+      priority: 'danger',
+      onAction: onDelete,
+    },
+  ];
+
+  const menuOptions = hasUnsavedChanges
+    ? [
+        {
+          key: 'changed',
+          children: hasUnsavedChangesMenuOptions,
+        },
+        {
+          key: 'default',
+          children: defaultMenuOptions,
+        },
+      ]
+    : defaultMenuOptions;
+
+  return (
+    <TriggerIconWrap>
+      <StyledDropdownMenu
+        position="bottom-start"
+        triggerProps={{
+          size: 'zero',
+          showChevron: false,
+          borderless: true,
+          icon: (
+            <Fragment>
+              <StyledDropdownButton
+                {...triggerProps}
+                aria-label="Tab Options"
+                borderless
+                size="zero"
+                icon={<IconCompactEllipsis />}
+              />
+              {hasUnsavedChanges && <ChangedAndUnsavedIndicator role="presentation" />}
+            </Fragment>
+          ),
+          style: {width: '18px', height: '16px'},
+        }}
+        items={menuOptions}
+        offset={[-10, 5]}
+      />
+    </TriggerIconWrap>
+  );
+}
+
+function IconCompactEllipsis() {
+  return (
+    <SvgIcon>
+      <circle cx="8" cy="8" r="1.11" />
```

Comment:
Is this different from the `IconEllipsis` we already have? If this is necessary, we probably want to either add a prop to the existing icon or add this to the icons folder

Reply:
Yup - the existing IconEllipsis had too much spacing between the dots, which made it look a little cramped inside the dropdown button. This one just squeezes them a little closer. Agreed that this should just be controlled with a prop rather than a whole new component. 

`<IconEllipsis/>` (existing component):
<img width="138" alt="image" src="https://github.com/user-attachments/assets/e689eac2-4390-4c0a-84b5-6615fdb052c2">

`<IconCompactEllipsis/>` (new component): 
<img width="138" alt="image" src="https://github.com/user-attachments/assets/001a6f39-7249-4318-b867-08a154e3b79b">